### PR TITLE
Blog App: Add forms #7

### PR DESCRIPTION
**Pull request by siddghosh108**
- [ ]  Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it in the upcoming projects.

- [ ] Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

- [ ] Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        <div class="form-group">
+          <%= form.label :text %>
+          <%= form.text_area :text, class: 'form-control' %>
+        </div>
+
+        <%= form.hidden_field :author_id, value: current_user.id %>
+
+        <div class="form-group mt-2">
+          <%= form.submit 'Create Post', class: 'btn btn-primary' %>
```
Comment (Olamarx) [inline app/views/posts/new.html.erb:29]: - [ ] While your code appears well written, this form does not submit, when I added data, it failed to submit it and as a result, I could not check the comment form aspect. Kindly double-check it.
image

This is the error I get in the console and it is as regards the current user, I will implore you to just use the User.first alone.
image

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    @current_user ||= User.first
+  end
+  helper_method :current_user
```
Comment (Olamarx) [inline app/controllers/application_controller.rb]: - [ ] I think we don't need to condition it here, all we need is the first user, just a demo. So I will encourage you to just have it as 
```
def current_user
  User.first
end
```

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @siddghosh108, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/new.html.erb (hunk 1 above)
Comment (siddghosh108) [reply app/views/posts/new.html.erb:29]: I have fixed the error and you can succesfully post now
Screenshot 2023-12-14 at 2 53 38 PM

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (siddghosh108) [reply app/controllers/application_controller.rb]: class ApplicationController < ActionController::Base
  def current_user
    User.first
  end
  helper_method :current_user
end

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @siddghosh108👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job with forms. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] Good job fixing the issues mentioned by the previous reviewer but I could not test the fixed issues because you deleted the migration files in the previous commits. Please make sure to add the migration files so the next code reviewer can test the project.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — COMMENTED** [review]
Hi @siddghosh108,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly rectify your linter errors.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by julie-ify — APPROVED** [review]
Hi @siddghosh108,

## Highlights!
- Good code organization and working forms✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**